main/lsp/requests/completion.cc

```diff
@@ -681,28 +681,6 @@ unique_ptr<CompletionItem> trySuggestSig(LSPTypecheckerInterface &typechecker,
         return nullptr;
     }
 
-    core::ClassOrModuleRef receiverSym;
-    if (core::isa_type<core::ClassType>(receiverType)) {
-        auto classType = core::cast_type_nonnull<core::ClassType>(receiverType);
-        receiverSym = classType.symbol;
-    } else if (auto appliedType = core::cast_type<core::AppliedType>(receiverType)) {
-        receiverSym = appliedType->klass;
-    } else {
-        // receiverType is not a simple type. This can happen for any number of strange and uncommon reasons, like:
-        // x = T.let(self, T.nilable(T::Sig));  x.sig {void}
-        return nullptr;
-    }
-
-    if (receiverSym == core::Symbols::rootSingleton()) {
-        receiverSym = core::Symbols::Object().data(gs)->lookupSingletonClass(gs);
-    }
-    auto methodOwner = targetMethod.data(gs)->owner;
-
-    if (!(methodOwner == receiverSym || methodOwner == receiverSym.data(gs)->attachedClass(gs))) {
-        // The targetMethod we were going to suggest a sig for is not actually in the same scope as this sig.
-        return nullptr;
-    }
-
```

Comment:
Is this all redundant after the changes to `NextMethodFinder`?

Reply:
Yes. I had to do this in NextMethodFinder for two reasons:

- I wanted to share this logic between suggested sigs and suggested YARD docs
- This logic relies on looking at the `enclosingMethod` that comes back in the LSP QueryResponse to figure out whether the method we find is owned by the same owner as the `enclosingMethod.data(gs)->owner`

For the YARD snippet case, we don't have any QueryResponses, which means no enclosing method, which means we need a purely syntactic way to check that we're in the same nesting scope as the queryLoc.